torch/optim/adam.py

```diff
@@ -198,7 +197,7 @@ def _init_group(
                     state['step'] = (
                         torch.zeros((1,), dtype=torch.float, device=p.device)
                         if self.defaults['capturable'] or self.defaults['fused']
-                        else torch.tensor(0.)
+                        else torch.tensor(0., dtype=torch.float64)
```

Comment:
Note that this won't change older saved states on optimizers. It might be simpler to keep this the same and change the dtype within the step?